### ISSUE TITLE
Allow to configure kubernetes context passing parameters

### ISSUE THREAD

**Issue by mcanevet** [oboukili/terraform-provider-argocd#65]
Unless I'm wrong, currently the only way to configure the provider to use it with port-forward is to specify a kubernetes context already set up in the `~/.kube/config` file. It would be nice to be able to pass kubernetes configuration just like they do in the helm provider for example.

Being able to be that would be great:
```hcl
provider "argocd" {
  kubernetes {
    host     = "https://104.196.242.174"
    username = "ClusterMaster"
    password = "MindTheGap"

    client_certificate     = file("~/.kube/client-cert.pem")
    client_key             = file("~/.kube/client-key.pem")
    cluster_ca_certificate = file("~/.kube/cluster-ca-cert.pem")
  }
}
```

**Comment by oboukili** [comment]
Hi @mcanevet, you might be right I never used the feature, only exposed what ArgoCD client provides and exposed it through the provider, let me take a look further this weekend.

PS: that's funny I used to live in Chambéry up until a year ago, small world :)

**Comment by oboukili** [comment]
You are right, ArgoCD added more options to the API client since the inception of the provider (around ArgoCD 1.4). I can add the additional options [1]. One thing to note is that you still have to pass a token instead of the User/Password parameter. If that's really important to you I can add the feature to get a token from the API Server in a two-steps process.
It will just take a little more time to add proper tests, compared to simply exposing the API Client parameters to the provider options.

[1]
```go
// ClientOptions hold address, security, and other settings for the API client.
type ClientOptions struct {
	ServerAddr           string
	PlainText            bool
	Insecure             bool
	CertFile             string
	ClientCertFile       string
	ClientCertKeyFile    string
	AuthToken            string
	ConfigPath           string
	Context              string
	UserAgent            string
	GRPCWeb              bool
	GRPCWebRootPath      string
	PortForward          bool
	PortForwardNamespace string
	Headers              []string
}
```

**Comment by oboukili** [comment]
Ok nevermind I completely misunderstood your request, you were talking about adding Kubernetes-level parameters instead of a context. I will see what I can do.

**Comment by raphink** [comment]
This should allow to pass Kubernetes config overrides https://github.com/argoproj/argo-cd/pull/5906

**Comment by fvdnabee** [comment]
+1 for passing the kubernetes configuration as an input to the argocd provider, it would help to authenticate with the k8s provider in the same way as the [terraform tutorial on EKS](https://github.com/hashicorp/learn-terraform-provision-eks-cluster/blob/master/kubernetes.tf):

```terraform
provider "argocd" {
  kubernetes {
    host                   = data.aws_eks_cluster.cluster.endpoint
    cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
    exec {
      api_version = "client.authentication.k8s.io/v1alpha1"
      args        = ["eks", "get-token", "--cluster-name", data.aws_eks_cluster.cluster.name]
      command     = "aws"
    }
  }
}
```

**Comment by raphink** [comment]
Since https://github.com/argoproj/argo-cd/pull/5906 was merged, it is now possible to pass a `KubeOverrides` field in the `ClientOptions` structure.

We can thus improve the `initApiClient()` method to pass Kubernetes overrides. I'll add that once #75 is merged, to avoid conflicts.

**Comment by fredleger** [comment]
if i'm not wrong because of this one we can't have a kube config file (or multiples kube config files) in non standard places like the terraform plan folder even referenced correctly by the KUBECONFIG variable ?

Would be nice to use this as well (it can be a different new issue)  like others providers does ex: helm / kubernetes official ones

**Comment by raphink** [comment]
I've added a draft for this feature at https://github.com/oboukili/terraform-provider-argocd/pull/100. I will rebase it once #75 is done.
